src/Hooks/useAuthActions.js

```diff
@@ -91,6 +100,7 @@ export default function useAuthActions() {
       const provider = new GoogleAuthProvider();
       const userCredential = await linkWithPopup(auth, provider);
       await onRegistrationSuccess(userCredential, "google");
+      await writeAllWatchlistDataEntries();
```

Comment:
Is there a way to merge the watchlist data writing logic into `onRegistrationSuccess`?  I'd like to keep the individual likeWith*/registerWith* functions below free of business logic.  Maybe something after the localuser data is saved like:

```js
// On a non-anonymous registration, write watchlist data for all watchlists.
if (isRegistration && authProvider !== "anonymous") {
  await writeAllWatchlistDataEntries();
}
```

Should be a no-op for first-time registrations, since their `lists` arrays should be empty, and so `await Promise.all([])` will immediately resolve.  Wdyt?

Reply:
Sounds good to me! For the no-op, I added a conditional statement checking whether `localUser.lists.length > 0`.